ibm-openbmc/bmcweb
# New property to always AllowUnauthACFUpload

Background: The Access Control File (ACF) service access is a 2-factor authentication scheme intended for BMC administrators to determine when authorized service and support representatives (SSRs) are allowed to access their BMC but not allow the administrator to have access. In this scheme, only the SSR can create valid ACFs,m which they then give it to the BMC admin for upload to the BMC.  Doing so authorizes the SSR to login to the BMC's service account via an unique password.

A problem with this scheme is some BMC admins lose their password.  To solve this, Operator Panel function 74 opens a 30 minute time window where anyone is allowed to upload a valid ACF to the BMC.

A problem with that solution is that in some installations there is difficulty gaining physical access to the operator panel.  To solve that, there is a new security setting to open up the ACF upload function...

This adds a new BMC security setting as a Redfish OEM property: URI /redfish/v1/AccountService/Accounts/service property Oem.IBM.ACF.AllowUnauthACFUpload can be set by the admin.  When true, any network agent is allowed to upload an ACF file to the BMC. The default value is false.  This capability is included in the existing URI /redfish/v1 property Oem.IBM.ACFWindowActive.  For example, when AllowUnauthACFUpload=true, ACFWindowActive is true.

Tested: via Redfish scenarios
1. Ensure AllowUnauthACFUpload=false by default.
2. Ensure a non admin user is not allowed to GET or PATCH AllowUnauthACFUpload.
3. Ensure the admin can PATCH AllowUnauthACFUpload true and false, and GET the correct values, and these values survive BMC reboot.
4. With (OpPanel fn 74 is not active) and AllowUnauthACFUpload=false, ensure ACFWindowActive=false.  And unauth ACF upload is not allowed.
5. With (OpPanel fn 74 is not active) and AllowUnauthACFUpload=true,

Cheatsheet:
```
busctl set-property com.ibm.PanelApp /com/ibm/panel_app com.ibm.panel ACFWindowActive b true
curl -k -H "X-Auth-Token: $TOKEN" -X PATCH -d '{"Oem":{"IBM":{"ACF":{"AllowUnauthACFUpload": true}}}}' https://${bmc}/redfish/v1/AccountService/Accounts/service
curl -k -H "X-Auth-Token: $TOKEN" -X GET https://${bmc}/redfish/v1
curl -k -X PATCH -d '{"Oem":{"IBM":{"ACF":{"ACFFile":"'${acf}'"}}}}' https://${bmc}/redfish/v1/AccountService/Accounts/service

```
